internal/services/cragcast/cragcast.go

```diff
@@ -0,0 +1,227 @@
+		return nil, err
+	}
+	cragcast := &Cragcast{
+		Days: []*DailyForecast{},
+	}
+
+	for _, hour := range hourlyForecasts {
+		cragcastDayIndex := len(cragcast.Days) - 1
+		// do we need a timezone conversion here?
+		day := hour.StartOfHour.Day()
+		if len(cragcast.Days) == 0 || cragcast.Days[cragcastDayIndex].Date.Day() != day {
+			// we are on a new day so we append a day to our cragcast
+			initialLowFahrenheit := float32(1000)
+			cragcast.Days = append(cragcast.Days, &DailyForecast{
+				Date:               hour.StartOfHour,
+				LowOfDayFahrenheit: &initialLowFahrenheit,
+				Hours:              []*HourlyForecast{},
+			})
+			cragcastDayIndex++
+		}
+
+		cragcastDay := cragcast.Days[cragcastDayIndex]
+		if hour.TotalPrecipMm != nil {
+			newPrecipMm := getPointerValueWithDefault(cragcastDay.TotalPrecipMm) + *hour.TotalPrecipMm
+			cragcastDay.TotalPrecipMm = &newPrecipMm
+		}
+
+		if hour.ChanceOfPrecip != nil && getPointerValueWithDefault(cragcastDay.ChanceOfPrecip) < *hour.ChanceOfPrecip {
+			cragcastDay.ChanceOfPrecip = hour.ChanceOfPrecip
+		}
+		if hour.TemperatureFahrenheit != nil && getPointerValueWithDefault(cragcastDay.LowOfDayFahrenheit) > *hour.TemperatureFahrenheit {
+			cragcastDay.LowOfDayFahrenheit = hour.TemperatureFahrenheit
+		}
+		if hour.TemperatureFahrenheit != nil && getPointerValueWithDefault(cragcastDay.HighOfDayFahrenheit) < *hour.TemperatureFahrenheit {
+			cragcastDay.HighOfDayFahrenheit = hour.TemperatureFahrenheit
+		}
+
+		if hour.Humidity != nil && getPointerValueWithDefault(cragcastDay.HumidityHigh) < *hour.Humidity {
+			cragcastDay.HumidityHigh = hour.Humidity
+		}
+
+		if hour.snowfallMm != nil {
+			newSnowfall := getPointerValueWithDefault(cragcastDay.snowfallMm) + *hour.snowfallMm
+			cragcastDay.snowfallMm = &newSnowfall
+		}
+
+		cragcastDay.Hours = append(cragcastDay.Hours, hour)
+	}
+
+	return cragcast, nil
+}
+
+func getProcessedHours(weatherClientForecast *noaaTypes.GridpointGeoJson) ([]*HourlyForecast, error) {
+	hours := []*HourlyForecast{}
+	// make a dictionary where times are the keys and values are hourly forecasts, loop through all sky cover, temp, precip etc...
+	// and add this info to each hour
+	// once that is done, loop through keys to make the hourly forecast and then sort by date
+
+	hoursMap := make(map[string]*HourlyForecast)
+
+	quantValuesToProcess := []QuantitativeValueType{TEMPERATURE, HUMIDITY, PRECIPITATION_PROBABILITY, QUANTITATIVE_PRECIPITATION, SKY_COVER, SNOWFALL_AMOUNT, WIND_DIRECTION, WIND_SPEED}
+
+	for _, quantValue := range quantValuesToProcess {
+		err := processQuantitativeValues(quantValue, weatherClientForecast.Properties.AdditionalProperties, hoursMap)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	for _, hourlyForecast := range hoursMap {
+		hours = append(hours, hourlyForecast)
+	}
+
+	sort.Slice(hours, func(i, j int) bool {
+		return hours[i].StartOfHour.Before(hours[j].StartOfHour)
+	})
+
+	return hours, nil
+}
+
+type QuantitativeValueType string
```

Comment:
**nit (non-blocking)**: As you've seen, go doesn't have true `enum`s (truly sad, I know), so I grok what you're trying to do here, but is perhaps less than ideal. ie maintaining `HUMIDITY` with `relativeHumidity` mapping. Also, when `QuantitativeValueType` is used as a parameter's type, it provides no guarantee since it is essentially a `string`.

Also, typically enums are "enumerated" using `iota`, as described in [Effective Go: Constants](https://go.dev/doc/effective_go#constants). Although this would break some of your switch statements lower down.

Overall, I'm _ok_ with what you've got here, but would implore you to consider whether there is any value in the `QuantitativeValueType`, as it is 🙂.

Hope that makes sense, more so something for you to noodle on as you're newer to Go, don't let this block progress though.